sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/implementation/websocket/WebSocketClientHandler.java

```diff
@@ -110,12 +121,61 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
                 // close initiated from client, client already sent CloseWebSocketFrame
                 ch.close();
             }
+        } else if (msg instanceof TextWebSocketFrame || msg instanceof ContinuationWebSocketFrame) {
+            processFrame(msg);
+        } else {
+            loggerReference.get().atWarning().addKeyValue("frameType", msg).log("Unknown WebSocketFrame type.");
+        }
+    }
+
+    private void processFrame(WebSocketFrame frame) {
```

Comment:
If we were to use `CompositeByteBuf` then, once the final fragment arrives, the BinaryData can be created from the `this.cbb`, then the `CompositeByteBuf` instance that `this.cbb` references can be released,  `this.cbb.release()`,  and a new `CompositeByteBuf` instance can be allocated and assigned to `this.cbb`